rfcs/proposed/host_backend_histogram/README.md

```diff
@@ -0,0 +1,165 @@
+histogram is for the number of elements in the input sequence to be far greater than the number of output histogram
+bins. We may be able to use that to our advantage.
+
+### Code Reuse
+Our goal here is to make something maintainable and to reuse as much as we can which already exists and has been
+reviewed within oneDPL. With everything else, this must be balanced with performance considerations.
+
+### unseq Backend
+Currently oneDPL relies upon openMP SIMD to provide its vectorization, which is designed to provide vectorization across
+loop iterations. OneDPL does not directly use any intrinsics which may offer more complex functionality than what is
+provided by OpenMP.
+
+As mentioned above, histogram looks to be a memory bandwidth-dependent algorithm. This may limit the benefit achievable
+from vector instructions as they provide assistance mostly in speeding up computation.
+
+For histogram, there are a few things to consider. First, lets consider the calculation to determine which bin to
+increment. There are two APIs, even and custom range which have significantly different methods to determine the bin to
+increment. For the even bin API, the calculations to determine selected bin have some opportunity for vectorization as
+each input has the same mathematical operations applied to each. However, for the custom range API, each input element
+uses a binary search through a list of bin boundaries to determine the selected bin. This operation will have a
+different length and control flow based upon each input element and will be very difficult to vectorize.
+
+Second, lets consider the increment operation itself. This operation increments a data dependant bin location, and may
+result in conflicts between elements of the same vector. This increment operation therefore is unvectorizable without
+more complex handling. Some hardware does implement SIMD conflict detection via specific intrinsics, but this is not
+generally available, and certainly not available via OpenMP SIMD. Alternatively, we can multiply our number of temporary
+histogram copies by a factor of the vector width, but we will need to determine if this is worth the overhead, memory
+footprint, and extra accumulation at the end. OpenMP SIMD does provide an `ordered` structured block which we can use to
+exempt the increment from SIMD operations as well. It must be determined if SIMD is beneficial in either API variety. It
+seems only possible to be beneficial for the even bin API, but more investigation is required.
+
+Finally, for our below proposed implementation, there is the task of combining temporary histogram data into the global
+output histogram. This is directly vectorizable via our existing brick_walk implementation.
+
+### Serial Backend
+We plan to support a serial backend for histogram APIs in addition to openMP and TBB. This backend will handle all
+policies types, but always provide a serial unvectorized implementation.
+
+## Existing Patterns
+
```

Comment:
If we intent to give some information about OneDPL parallel backend patterns on which histogram can based on, I would notify, there is not "count_if" pattern, there is "reduce"("transform_reduce") pattern.
When a man says "reduce", it becomes more or less obvious that histogram calculation based on reduce is not effective at all.

Reply:
I clarified the language a little here to make it more clear that copy_if uses reduce internally.  I still think it deserves some text describing it as it may not be immediately obvious to everyone that reduce is not well matched.